src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -3,8 +3,15 @@
 var React = require("react/addons");
 
 var AppComponent = React.createClass({
+  propTypes: {
+    router: React.PropTypes.object.isRequired
+  },
+
   onClick: function () {
-    this.props.onClick(this.props.model);
+    this.props.router.navigate(
+      "#apps" + this.props.model.get("id"),
```

Comment:
Yes, 2 good points.
Now using encodeURIComponent to allow route 'apps//sleep30', it becomes 'apps/%2Fsleep30'.
'/sleep30' is an appid.